src/catchup/CatchupRange.cpp

```diff
@@ -37,4 +39,45 @@ operator!=(CatchupRange const& x, CatchupRange const& y)
 {
     return !(x == y);
 }
+
+CatchupRange
+makeCatchupRange(ResolvedCatchupConfiguration const& configuration,
+                 HistoryManager const& historyManager)
+{
+    auto toCheckpoint =
+        historyManager.nextCheckpointLedger(configuration.toLedger() + 1) - 1;
+
+    auto firstNotKnownCheckpoint = [&]() {
+        auto lcl = configuration.lastClosedLedger();
+        auto nextCheckpoint = historyManager.nextCheckpointLedger(lcl + 1) - 1;
+        if (lcl == nextCheckpoint)
+        {
+            return nextCheckpoint + historyManager.getCheckpointFrequency();
+        }
+        return nextCheckpoint;
+    }();
+
+    auto catchupFromLedger =
+        configuration.toLedger() > configuration.count()
+            ? configuration.toLedger() - configuration.count()
+            : 0;
+    auto catchupFromCheckpoint =
+        historyManager.nextCheckpointLedger(catchupFromLedger) - 1;
+
+    // ensure that we have at least configuration.count() ledgers if
```

Comment:
this needs better comments/variable names, what is really happening here is that we're attempting to go to the previous checkpoint if the one currently computed brings us before catchupFromLedger (the highest ledger we should be starting from)

Reply:
I would phrase it as:

`we're attempting to go to the previous checkpoint if the one currently computed would make us apply less ledgers than configuration.count() `